src/app/test/test-folder-tree/test-folder-tree.component.ts

```diff
@@ -24,81 +26,59 @@ export class TestFolderTreeComponent {
     autoSelectCondition: (item: CreateTreeItem) => this.autoSelectItem(item),
     determineIconClass: SimpleFileTreeUtil.conditionalCssClass,
   };
-  originalItems?: CreateTreeItem[];
 
-  autoSelectItem(item: CreateTreeItem): boolean {
-    return item.name === this.rootFolder.name;
-  }
-
-  selectItem(value: string): void {
-    value = this.removeSlashesFromPathEnds(value);
-    const path: string = value.length > 0 ? `Reports/${value}` : 'Reports';
-    this.tree.selectItem(path);
+  setData(data: TestListItem[]): void {
+    this.convertToTree(data);
+    this.tree.selectItem(this.rootFolder.path!);
   }
 
-  removeSlashesFromPathEnds(path: string): string {
-    if (path.startsWith('/')) {
-      path = path.slice(1);
-    }
-    if (path.endsWith('/')) {
-      path = path.slice(0, -1);
+  convertToTree(data: TestListItem[]): void {
+    this.tree.items = [];
+    this.rootFolder.children = [];
+    for (let item of data) {
+      if (item.path) {
+        this.nextAction(
+          item,
+          item.path.split('/').filter((s) => s !== ''),
+          this.rootFolder,
+        );
+      }
```

Comment:
Isn't it better to do this recursively?
It gives the opportunity to actually track the parent and children and make it behave like an actual NodeTree structure, now you're doing it the opposite way making it quite harder for yourself

Reply:
How do you mean? `nextAction` is already called recursively

Reply:
You do use  recursion, but not for the children of the root folder.
Why not use recursion for every folder starting with the root folder itself? Next action already tries to get all children recursively so why resort to also using a for-loop for what is essentially the same logic but closer to the root?

Reply:
Because it makes folders based on the path of the items, the items are supplied in a 1 dimensional array, so they don't have a parent-child structure yet.

Reply:
I don't see how that changes anything either way, but you're already splitting the path into route parts right?
`/test/test-a/a` and `/test/test-b/b` would be split up into a root folder with both folders `test-a` and `test-b` and than that recursively deeper right?
What does the for-loop do any different except actually calling the array.split function?

Reply:
Here is an example of how I would solve it, after talking with Sergi about it
https://www.typescriptlang.org/play/?#code/C4TwDgpgBAlsEFsoF4oG8BQVtQHYEMEIAuKAZ2ACcZcBzAGixzH2AAtSLq7GcoBjNjAA2AE0oRcpOIgDaAXQDcGAL4YM-APa4KUSps3Bp8JKjQEipAOT7DV+lBbtrAensChYiVKgKVyjW1dYRhdVFkmbCsISRdgAHcYl3EYCBcIGlEIe0ioaNiEpIzcLLdeHCt02OKsnL583BcmqsbCtJrs8qiOqwx5dQAzTUooAAotHWBHVjYoTQGoEIoASnRcid0yfFw4GAAvCFEASRMUcm3dg4AFGdGnNmXc-FFjkzIAMX0EACUIYVYYAA3CA3dijLY7YD7Q4nRAAOjIYBCwFGlSsywctmAjzUgR0mmEEDhwk0tFGWMeGAGAFdcPwodpzpDoaC2HcZpwqDRaMtOdxaGs+BJgNTKLhpuw4RIkfh+BBRi4ADouADULgcaOWUogMrlCoAesqAD7KgAk6ry6OUuJpdIZ4uer0QHy+v3+UOBrPJfwBno55C5dAUDhY3iMsBMvKggM0MFEgpwMAW3vdQJBM2Jklo7BQyFQAAZVpg+EKICKxco+Li+BsprgIPEAMKeUTGRBnYsl7AWEh6H0e9PsWT5+QOXJ8e6kABWsdwbt9g7ZockwDh90x-bTrOH8gx45wghE4kkpAUuTUNaCUwgAA9QlC6M2j2dl7hV4evJIAPxwgaZUbjC2qzIAAfB4R5wj2uZ9qmfpDiOjyXpM4FiPOA6smcEiwYuCIhHqACMiGJsmACEt73tyT5iEW+7YK+74tt4a7UmQbL1k2QFQE0AjbFA+DCBIzwgFAoiaOQNByhGZB9gMEDeJJoTTGQZCHLA4rsNAIT1lAABGfyaPEfHSYkwjCLRfblg6LywggLqaD8m5wWyH6iGhW4zA47FUaIRHYBeODCqKVlOrZnz2W5TmAUeEWLg45EUJRQGqIMtL0jAjIzjQMVen+lAUKyfLcg4KkTKIBUBvyNGlpZYy5flGaSKIZAAOpwGympQF+lpQK46JQCq5AQKVrKqEAA

Reply:
Fixed it.